docs/NamingConvention.md

```diff
@@ -1,37 +1,61 @@
 - Max length: `1-63` chars
 - Start with lower case character, can’t end in `-` or `_` or any special char
 > Preferred names have all lower case
 >> A preferred name can be regexed by `[a-z]([-a-z0-9]*[a-z0-9])?`
 
 ## Azure resource names
-Azure resources will have four parts:
-- `[ prefix ]-( project-short-form )-[ identifier ]-[ suffix ]-( environment-type )`
-  - Mandatory: `(project-short-form)`. Example: `ngsa`, `sp`
-  - Mandatory: `(environment-type)` - usage env. Possible val: `prod`, `test` and `dev`
-  - Optional: `[prefix]` or `[suffix]` - resource type suffix. Follows [Azure naming convention][1]
-    - Common resource prefix/suffix:
-        | Az Resource Name   | Prefix/Suffix |
-        |--------------------|----------|
-        | Key Vault          | kv-      |
-        | Function           | func-    |
-        | Virtual Machine    | vm-      |
-        | VM Scale Set       | vmss-    |
-        | CosmosDB           | cosmos-  |
-        | AKS Cluster        | aks-     |
-        | Container Registry | cr-      |
-  - Optional: `[identifier]` - refers to descriptive label
-  - Each part will be separated by a hyphen `-`
-  - Resource names will use either suffix or prefix…
-
-## Naming convention for Kubernetes resources 
-### Label format
-
-## Resourcese
+Azure resources consists of four parts: `( project-short-form )-[ identifier ]-( environment-type )-[ suffix ]`
+  - Mandatory: `(project-short-form)`.
+  - Mandatory: `(environment-type)` - usage env.
+    - Possible val: `pre`, `test`, `stage`, `prod`, and `dev` [?]
+      - `pre` is pre-production
+  - Optional: `[identifier]` - refers to descriptive label.
+  - Best effort to fit based on specific resource naming constraints.
+  - Each part will be separated by a hyphen `-`.
+  - Azure resource suffixes can be found [here][1].
+  - Examples:
+    - `ngsa-grafana-dev-kv`
+    - `ngsa-istio-pre-cosmos`
+    - `ngsa-grafana-prod-appi`
+
+## Kubernetes Labels
+Labels are key/value pairs. Valid label keys have two segments: an optional *prefix* and *name*, separated by a slash (/) (e.g `helm.sh/chart`).
+
+Prefixes are optional, and ensures that recommended labels do not get mixed up with private labels.
+
+If prefix is omitted, the label is considered to be private to the user.
+
+Detailed syntax and requirements can be found at: ["Kubernetes label syntax and charset"][4].
+
+Adhering to ["Labels recommended by kubernetes"][3], following labels are required for essential kubernetes objects (pods, deployments, services, hpa etc.):
+
+- app.kubernetes.io/name: name-of-the-component
+- app.kubernetes.io/version: version-of-component
+- app.kubernetes.io/component: type-of-the-component
+
+Examples:
+- app.kubernetes.io/name=ngsa
+- app.kubernetes.io/version=1.16.0
+- app.kubernetes.io/component=app
+- app.kubernetes.io/component=database
+
+## Kubernetes Namespaces
+K8s namespace will follow the format: `[Team Name]-[Application Name]`
+
+Examples:
```

Comment:
team name seems redundant to me. how many teams have the same app name?

Reply:
This is to align with the customer. They use a team name prefix in their namespaces.